File: UI/musicshop_admin/lib/models/amplifier/amplifier.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:musicshop_admin/models/Abstract/product.dart';
import 'package:musicshop_admin/models/brand/brand.dart';

part 'amplifier.g.dart';

@JsonSerializable()
class Amplifier {
  int? id;
  String? productNumber;
  Brand? brand;
  String? model;
  double? price;
  String? description;
  int? voltage;
  int? powerRating;
  bool? headphoneJack;
  bool? usbjack;
  int? numberOfPresets;
  DateTime? createdAt;
  DateTime? updatedAt;

  Amplifier();

  factory Amplifier.fromJson(Map<String, dynamic> json) =>
      _$AmplifierFromJson(json);

  Map<String, dynamic> toJson() => _$AmplifierToJson(this);
}

File: MusicShop.Model/Amplifier.cs
﻿using MusicShop.Model.BaseModel;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicShop.Model
{
    public class Amplifier : Product
    {
        public string? ProductNumber { get; set; }
        public int? ProductImageId { get; set; }
        public int? BrandId { get; set; }
        public string? Model { get; set; }
        public decimal? Price { get; set; }
        public string? Description { get; set; }
        public int? Voltage { get; set; }
        public int? PowerRating { get; set; }
        public bool? HeadphoneJack { get; set; }
        public bool? Usbjack { get; set; }
        public int? NumberOfPresets { get; set; }
    }
}
